accord-core/src/main/java/accord/local/Node.java

```diff
@@ -268,6 +268,33 @@ public <T> AsyncChain<T> withEpoch(long epoch, Supplier<? extends AsyncChain<T>>
         }
     }
 
+    public AsyncChain<?> awaitEpoch(long epoch)
+    {
+        if (topology.hasEpoch(epoch))
+            return AsyncResults.SUCCESS_VOID;
+        CommandStore executingOn = CommandStore.maybeCurrent();
+        configService.fetchTopologyForEpoch(epoch);
+        AsyncChain<Void> chain = topology.awaitEpoch(epoch);
+        if (executingOn != null)
+            chain = chain.withExecutor(executingOn);
+        return chain;
+    }
+
+    public AsyncChain<?> awaitEpoch(@Nullable EpochSupplier epochSupplier)
```

Comment:
this is a weird api, but `execute_at` is so common that felt this was good to have... when we don't know if we know execute_at yet... we should call this to make sure its safe to make progress...

Reply:
The only method using this is not called from anywhere. Do we really need both the `withEpoch` approach which can be combined with checking if the epoch is already present to get the same effect?

Reply:
This looks like a rebase issue, the method should have `BiConsumer<? super Known, Throwable> callback` but that got lost in the rebase and became `BiConsumer<Known, Throwable> callback` which made the partial route one called 100% of the time... once I fixed the signature then this is called from SPL